### PR TITLE
Add custom image creation

### PR DESCRIPTION
**Pull request by VictorAvelar**
**Summary of changes**

- Added Create method to ImageService interface
- Added method implementations
- Added missing `ArgImage_` constants
- Extended test implementation to comply with the aforementioned interface
- Regenerated mocks
- Added tests for command execution

Attempts to Solve: #367

### CODE REVIEW

**Comment by zachgersh** [comment]
Hey @VictorAvelar - nice job here! Would you mind also taking a shot at adding an integration test (if you feel comfortable)?

Here's an example to get you going (more in that directory):
https://github.com/digitalocean/doctl/blob/master/integration/droplet_create_test.go

**Comment by VictorAvelar** [comment]
> Hey @VictorAvelar - nice job here! Would you mind also taking a shot at adding an integration test (if you feel comfortable)?
> 
> Here's an example to get you going (more in that directory):
> https://github.com/digitalocean/doctl/blob/master/integration/droplet_create_test.go

Sure 👌

File: integration/image_create_test.go [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+				w.WriteHeader(http.StatusUnauthorized)
+				return
+			}
+
+			var err error
+			_, err = ioutil.ReadAll(req.Body)
+			expect.NoError(err)
+
+			w.Write([]byte(imageCreateResponse))
+			return
```
Comment (zachgersh) [inline integration/image_create_test.go]: needs a default case like here (makes the test fail if there is an unhandled path): 
https://github.com/digitalocean/doctl/blob/34749b0f09bde81a41097fedd5e724a36dc7983c/integration/droplet_get_test.go#L35-L42

server should handle the path (this currently returns the same response on every call):
https://github.com/digitalocean/doctl/blob/34749b0f09bde81a41097fedd5e724a36dc7983c/integration/droplet_get_test.go#L26

File: integration/image_create_test.go [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+		}))
+	})
+
+	when("all the required flags are passed", func() {
+		cmd := exec.Command(builtBinaryPath,
+			"--image-name", "ubuntu-18.04-minimal",
+			"--image-url", "http://cloud-images.ubuntu.com/minimal/releases/bionic/release/ubuntu-18.04-minimal-cloudimg-amd64.img",
```
Comment (zachgersh) [inline integration/image_create_test.go]: really appreciate using a real image url 💃

File: integration/image_create_test.go [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+		expect = require.New(t)
+		httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, req *http.Request) {
+			auth := req.Header.Get("Authorization")
+			if auth != "Bearer some-magic-token" {
+				w.WriteHeader(http.StatusUnauthorized)
+				return
+			}
+
+			var err error
+			_, err = ioutil.ReadAll(req.Body)
```
Comment (zachgersh) [inline integration/image_create_test.go]: This body does not need to be read since it is not used in an expectation.

**Comment by zachgersh** [comment]
@VictorAvelar - also have to make sure to register you integration test with the suite here:
https://github.com/digitalocean/doctl/blob/4e6ba07e932ece3a1d6d124df2db9b543ac1fb18/integration/init_test.go#L50

**Comment by VictorAvelar** [comment]
@zachgersh sorry for all the commits, it is my first time contributing, I was not familiar with the integration suite.

**Comment by zachgersh** [comment]
@VictorAvelar - no need to apologize! We really appreciate all the help and that you even took a shot at adding an integration test. A+++

**Comment by MattIPv4** [comment]
Hey, @VictorAvelar - Thanks a ton for this sweet PR! 😄

Would you please shoot me an email when you get a chance?
mcowley at digitalocean dot com 🎉

**Comment by VictorAvelar** [comment]
> Thanks a ton for this @VictorAvelar, really great work 🙂
> 
> I'm going to merge this as-is, but I'm going to open a follow up PR to make two slight modifications before we release. Those changes are,
> 
>     * Changing the custom image name to be a positional argument,
> 
>     * Adding a default of `Unknown` to the custom image distribution argument, as that's what our API expects as a default/fallback.
> 
> 
> Again, thanks for all your hard work. This is very high quality code!

@bentranter Thanks for the props, it was a pleasure and overall an amazing learning experience to contribute to your repo.

**Pull request merged**